src/utils/COMETAdapter.cpp

```diff
@@ -0,0 +1,577 @@
+  {
+   vector<ResidueModification> modifications;
+
+    // iterate over modification names and add to vector
+    for (StringList::iterator mod_it = modNames.begin(); mod_it != modNames.end(); ++mod_it)
+    {
+      String modification(*mod_it);
+      modifications.push_back(ModificationsDB::getInstance()->getModification(modification));
+    }
+
+    return modifications;
+  }
+
+  void removeTempDir_(const String& tmp_dir)
+  {
+    if (tmp_dir.empty()) return; // no temp. dir. created
+
+    if (debug_level_ >= 2)
+    {
+      writeDebug_("Keeping temporary files in directory '" + tmp_dir + "'. Set debug level to 1 or lower to remove them.", 2);
+    }
+    else
+    {
+      if (debug_level_ == 1) writeDebug_("Deleting temporary directory '" + tmp_dir + "'. Set debug level to 2 or higher to keep it.", 1);
+      File::removeDirRecursively(tmp_dir);
+    }
+  }
+
+  void createParamFile_(ostream& os)
+  {
+    os << "# comet_version 2016.01 rev. 2\n";               //required as first line in the param file
+    os << "# Comet MS/MS search engine parameters file.\n";
+    os << "# Everything following the '#' symbol is treated as a comment.\n";
+    os << "database_name = " << getStringOption_("database") << "\n";
+    os << "decoy_search = " << 0 << "\n"; // 0=no (default), 1=concatenated search, 2=separate search
+    os << "num_threads = " << getIntOption_("threads") << "\n";  // 0=poll CPU to set num threads; else specify num threads directly (max 64)
+
+    // masses
+    os << "peptide_mass_tolerance = " << getDoubleOption_("precursor_mass_tolerance") << "\n";
+    os << "peptide_mass_units = " << getIntOption_("precursor_error_units") << "\n";                  // 0=amu, 1=mmu, 2=ppm
+    os << "mass_type_parent = " << getIntOption_("mass_type_parent") << "\n";                    // 0=average masses, 1=monoisotopic masses
+    os << "mass_type_fragment = " << getIntOption_("mass_type_fragment") << "\n";                  // 0=average masses, 1=monoisotopic masses
+    os << "precursor_tolerance_type = " << getIntOption_("precursor_tolerance_type") << "\n";            // 0=MH+ (default), 1=precursor m/z; only valid for amu/mmu tolerances
+    os << "isotope_error = " << getIntOption_("isotope_error") << "\n";                      // 0=off, 1=on -1/0/1/2/3 (standard C13 error), 2= -8/-4/0/4/8 (for +4/+8 labeling)
+
+    // search enzyme
+
+    // TODO: complete
+    map<String, Size> map_oms2comet;
+    map_oms2comet["Trypsin"] = 1;
+    map_oms2comet["Arg-C"] = 5;
+    map_oms2comet["Asp-N"] = 6;
+    map_oms2comet["Chymotrypsin"] = 10;
+    map_oms2comet["CNBr"] = 7;
+    map_oms2comet["Lys-C"] = 3;
+    map_oms2comet["PepsinA"] = 9;
+    map_oms2comet["Trypsin/P"] = 2;
+    map_oms2comet["no cleavage"] = 0;   
+
+    String enzyme_name = getStringOption_("enzyme");
+    Size enzyme_number = 1;
+    if (map_oms2comet.find(enzyme_name) != map_oms2comet.end())
+    {
+      enzyme_number = map_oms2comet.at(enzyme_name);
```

Comment:
use `map_oms2comet[enzyme_name]`.
`.at()` is a gcc extension and not supported on some VS compilers